new-pipeline/src/evo.js

```diff
@@ -393,81 +384,38 @@ $(function () {
-              realBuckets);
-            multiselectSetOptions($("#aggregates"), newAggregates, [
-              newAggregates[0][0]]);
-          } else if(realKind == "categorical") {
-            var newAggregates = realBuckets.map((r, i) => {return [i.toString(), r]})
-            multiselectSetOptions($("#aggregates"), newAggregates, [
-              newAggregates[0][0]]);
-          }  else if (realKind === "boolean" || realKind === "flag") {
-            var newAggregates = getHumanReadableBucketOptions(realKind,
-              realBuckets);
-            multiselectSetOptions($("#aggregates"), newAggregates, [
-              newAggregates[0][0]]);
-
-            // Boolean histograms should always start off with all options selected
-            $("#aggregates")
-              .multiselect("selectAll", false)
-              .multiselect("updateButtonText");
-          } else { // realKind is another histogram kind, or null because we didn't have any data
-            var newAggregates = gDefaultAggregates.map(function (entry) {
-              return [entry[0], entry[1]];
-            });
-
-
-            multiselectSetOptions($("#aggregates"), newAggregates, ["median", "5th-percentile", "95th-percentile"]);
-          }
-
-          // Load aggregates from state on first load
-          newAggregates = newAggregates.map(function (entry) {
-            return entry[0];
-          })
-          aggregates = gInitialPageState.aggregates.filter(function (
-            aggregate) {
-            return newAggregates.indexOf(aggregate) >= 0;
-          });
-          if (!gLoadedAggregatesFromState && aggregates.length > 0) {
-            gLoadedAggregatesFromState = true;
-            $("#aggregates")
-              .multiselect("deselectAll", false)
-              .multiselect("select", aggregates);
-          }
+function updateAggregates(kind, buckets) {
+  gCurrentKind = kind;
+  let newAggregates;
+  if (kind === "enumerated") {
+    newAggregates = getHumanReadableBucketOptions(kind, buckets);
+    multiselectSetOptions($("#aggregates"), newAggregates, [newAggregates[0][0]]);
+  } else if(kind == "categorical") {
+    newAggregates = buckets.map((r, i) => {return [i.toString(), r]})
+    multiselectSetOptions($("#aggregates"), newAggregates, [newAggregates[0][0]]);
+  }  else if (kind === "boolean" || kind === "flag") {
+    newAggregates = getHumanReadableBucketOptions(kind, buckets);
+    multiselectSetOptions($("#aggregates"), newAggregates, [newAggregates[0][0]]);
+
+    // Boolean histograms should always start off with all options selected
+    $("#aggregates")
+      .multiselect("selectAll", false)
+      .multiselect("updateButtonText");
+  } else { // `kind` is another histogram kind, or null because we didn't have any data
+    newAggregates = gDefaultAggregates.map(entry => [entry[0], entry[1]]);
+
+    multiselectSetOptions($("#aggregates"), newAggregates, ["median", "5th-percentile", "95th-percentile"]);
```

Comment:
These are the default selected aggregates, right? Consider moving it to a constant.